names/names.py

```diff
@@ -10,11 +10,14 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
+
+
 duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+# for name_1 in names_1:
+#     for name_2 in names_2:
+#         if name_1 == name_2:
+#             duplicates.append(name_1)
+duplicates = [elem for elem in names_1 if elem in names_2]
```

Comment:
Hmmm what's the run time on this? Even though this is comprehension one line. Make sure you didn't just rewrite the original just in a different way.

There are a few different ways that you could.

One of the way that you could do it, is turn `names_2` into a set so you can look up O(1) time

```
duplicates = []
names_2_set = set(names_2) #O(1)
for name in names_1: #O(n)
  if name in names_2_set: #O(1)
    duplicates.append(name) #O(1)
```
So this run time would be O(n)

I believe yours would still be O(n^2). Let me know if I am wrong